tests/functional/unit_testing/test_unit_testing.py

```diff
@@ -118,5 +124,52 @@ def test_basic(self, project):
         results = run_dbt(["run"])
         assert len(results) == 3
 
+        # Select by model name
         results = run_dbt(["unit-test", "--select", "my_model"], expect_pass=False)
         assert len(results) == 5
+
+        # Test select by test name
+        results = run_dbt(["unit-test", "--select", "test_name:test_my_model_string_concat"])
+        assert len(results) == 1
+
+        # Select, method not specified
+        results = run_dbt(["unit-test", "--select", "test_my_model_overrides"])
+        assert len(results) == 1
+
+        # Select using tag
+        results = run_dbt(["unit-test", "--select", "tag:test_this"])
+        assert len(results) == 1
```

Comment:
What do you think about splitting these assertions into their own test - perhaps `test_selection`? They will still require the call to `run_dbt(["run"])` as well.